### ISSUE TITLE
Error: Page "/[locale]/[...rest]" is missing `generateStaticParams()` so it cannot be used with "output: export" config.

### ISSUE THREAD

**Issue by viphakorn** [amannn/next-intl#822]
### Description

Hi,

I Pulled the demo Next.js project from demo https://github.com/amannn/next-intl/tree/main/examples/example-app-router, encountering this error while static export:

```terminal
Error: Page "/[locale]/[...rest]" is missing "generateStaticParams()" for "output: export".
```

I know "/[locale]/[...rest]" can be remove but losing custom 404 not found page.
Any quick workaround or guidance on resolving this would be appreciated.

Thanks

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

null

### Reproduction description

Steps to reproduce:
1. Set up a Next.js project from demo https://github.com/amannn/next-intl/tree/main/examples/example-app-router.
2. install deps `pnpm i`
3. Configure "output: export" in the Next.js config
4. build `pnpm run build`

### Expected behaviour

get rid of "/[locale]/[...rest]" or can be export

**Comment by amannn** [comment]
Yep, dynamic routes, including the 404 matching, are [not supported for static exports](https://nextjs.org/docs/app/building-your-application/deploying/static-exports#unsupported-features). The docs were a bit outdated here, I've fixed this in https://github.com/amannn/next-intl/commit/f2eccf036388906b2758e3f98e1c420631ba22a0. Thanks for the hint!
